### ISSUE TITLE
[BUG] Generic Provider: No verbose / debug logs are available in base capability functions

### ISSUE THREAD

**Issue by pranavgaikwad** [konveyor/analyzer-lsp#519]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

latest

### Priority

Minor

### Current Behavior

There are no detailed logs about symbols received in base capabilities in the generic provider. This is because loggers are not passed down to these functions.

### Expected Behavior

With a higher verbosity set on CLI, we should be able to see language server query and responses. 

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Run rule-example.yaml file with a higher verbosity level (10+) 

Notice that java / builtin prints symbols and query information but not the Python / Go provider.

### Environment

```markdown
- **Analyzer Image SHA** latest
- **JDTLS  base image SHA** latest
- **Provider Settings JSON** latest
```

**Comment by shawn-hurley** [comment]
```
UNKN[0048]                                               logger=go provider=grpc
UNKN[0048] Request received:                             logger=go provider=grpc
UNKN[0048] - ID: {<nil>}                                 logger=go provider=grpc
UNKN[0048] - Method: window/logMessage                   logger=go provider=grpc
UNKN[0048] - Params: {"type":3,"message":"2024/03/22 13:34:47 go/packages.Load #1: updating metadata for 196 packages\n"}  logger=go provider=grpc
UNKN[0048]                                               logger=go provider=grpc
UNKN[0049] Request received:                             logger=go provider=grpc
UNKN[0049] - ID: {<nil>}                                 logger=go provider=grpc
UNKN[0049] - Method: window/showMessage                  logger=go provider=grpc
UNKN[0049] - Params: {"type":3,"message":"Finished loading packages."}  logger=go provider=grpc
UNKN[0049]                                               logger=go provider=grpc
UNKN[0049] Request received:                             logger=go provider=grpc
UNKN[0049] - ID: {<nil>}                                 logger=go provider=grpc
UNKN[0049] - Method: window/logMessage                   logger=go provider=grpc
UNKN[0049] - Params: {"type":1,"message":"2024/03/22 13:34:47 tidy: diagnosing file:///analyzer-lsp/examples/golang/go.mod: err: exit status 1: stderr: go: downloading github.com/google/go-cmp v0.5.5\ngo: downloading sigs.k8s.io/yaml v1.2.0\ngo: downloading github.com/stretchr/testify v1.7.0\ngo: downloading github.com/davecgh/go-spew v1.1.1\ngo: downloading github.com/spf13/pflag v1.0.5\ngo: downloading gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 tested by\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1.test imports\n\tgithub.com/google/go-cmp/cmp: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/apis/meta/v1 tested by\n\tk8s.io/apimachinery/pkg/apis/meta/v1.test imports\n\tsigs.k8s.io/yaml: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/runtime tested by\n\tk8s.io/apimachinery/pkg/runtime.test imports\n\tgithub.com/stretchr/testify/assert: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/runtime tested by\n\tk8s.io/apimachinery/pkg/runtime.test imports\n\tgithub.com/stretchr/testify/require: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/apis/meta/v1 imports\n\tk8s.io/apimachinery/pkg/api/resource tested by\n\tk8s.io/apimachinery/pkg/api/resource.test imports\n\tgithub.com/spf13/pflag: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/apis/meta/v1 imports\n\tk8s.io/apimachinery/pkg/labels tested by\n\tk8s.io/apimachinery/pkg/labels.test imports\n\tgithub.com/google/go-cmp/cmp/cmpopts: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/apis/meta/v1 tested by\n\tk8s.io/apimachinery/pkg/apis/meta/v1.test imports\n\tk8s.io/apimachinery/pkg/util/diff imports\n\tgithub.com/davecgh/go-spew/spew: module lookup disabled by GOPROXY=off\ngithub.com/konveyor/analyzer-lsp/examples/golang imports\n\tk8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1 imports\n\tk8s.io/apimachinery/pkg/runtime imports\n\tsigs.k8s.io/structured-merge-diff/v4/value imports\n\tgopkg.in/yaml.v2 tested by\n\tgopkg.in/yaml.v2.test imports\n\tgopkg.in/check.v1: module lookup disabled by GOPROXY=off\n\n"}  logger=go provider=grpc
```


After @JonahSussman refactor, I think that this just needs to be cleaned up. We are printing a lot of logs here now based on the output above.
